File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares


func SquareOfSum(n int) int {
	var sum int 
	for num := 1; num <= n; num++ {
		sum += num
	}
	return sum * sum
}

func SumOfSquares(n int) int {
	var sum int
	for num := 1; num <= n; num++ {
		sum += num * num
	}
	return sum
}

func Difference(n int) int {
	var squareOfSum = SquareOfSum(n)
	var sumOfSquares = SumOfSquares(n)
	return squareOfSum - sumOfSquares
}

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"
#include <cmath>
#include <iostream>
using namespace std;

namespace difference_of_squares {

// TODO: add your solution here
    int square_of_sum(int num) {
        int result{};
        for (int i = 1; i <= num; i++) {
            result += i;
        }
        return result * result;
    }
    
    int sum_of_squares(int num) {
        int result{};
        for (int i = 1; i <= num; i++) {
            result += i * i;
        }
        return result;
    }
    
    int difference(int num) {
        return square_of_sum(num) - sum_of_squares(num);
    }

}  // namespace difference_of_squares
